dags/utils/slack_operator.py

```diff
@@ -87,29 +93,55 @@ def task_fail_slack_alert_critical(context):
 
 
 def task_fail_slack_alert(context):
+
+    task_instance = context.get("task_instance")
+    task = context.get("task")
+    exception = context.get("exception")
+    exception_type = type(exception).__name__ if exception else "Unknown"
+    exception_message = (
+        str(exception) if exception else "No exception message available"
+    )
+
+    # Extract additional information
     dag = context.get("dag")
+    dag_id = task_instance.dag_id
+    task_id = task_instance.task_id
+    exec_date = get_central_time_exec_data(context)
+    log_url = task_instance.log_url
+    duration = getattr(task_instance, "duration", "Not available")
+
     schedule_interval = dag.schedule_interval if dag else None
+
     schedule_description = format_schedule(schedule_interval)
 
-    slack_msg = """
-            :red_circle: Task Failed. 
-            *Task*: {task}  
-            *DAG*: {dag} 
-            *Schedule*: {schedule_description}
-            *Execution Time*: {exec_date}  
-            *Log URL*: {log_url} 
-            """.format(
-        task=context.get("task_instance").task_id,
-        dag=context.get("task_instance").dag_id,
-        schedule_description=schedule_description,
-        exec_date=get_central_time_exec_data(context),
-        log_url=context.get("task_instance").log_url,
-    )
+    byline = getattr(dag, "byline", None)
```

Comment:
I [tested this with no byline](https://austininnovation.slack.com/archives/C013G4RNJ01/p1739300608754489) and it adds this `None` at the top of the error message, replacing `None` with an empty string `""` should clean this up for the DAGs where we don't have a byline.

Reply:
Great suggestion. https://github.com/cityofaustin/atd-airflow/pull/268/commits/8176eccf3827cdbdc292b6ed61cc5fddce8e8004